nucypher/cli/painting.py

```diff
@@ -173,36 +182,149 @@ def paint_known_nodes(emitter, ursula) -> None:
+    staking_agent = ContractAgency.get_agent(StakingEscrowAgent, registry=registry)
+    policy_agent = ContractAgency.get_agent(PolicyManagerAgent, registry=registry)
+    adjudicator_agent = ContractAgency.get_agent(AdjudicatorAgent, registry=registry)
 
-    contract_payload = f"""
-| NuCypher Contracts |
 
-Chain .................... {blockchain.client.chain_name}
-Provider URI ............. {blockchain.provider_uri}
-Registry Path ............ {blockchain.registry.filepath}
+    try:
+        registry_filepath = registry.filepath
+    except AttributeError:
+        registry_filepath = "No filepath"
 
+    contracts = f"""
+| Contract Deployments |
 NucypherToken ............ {token_agent.contract_address}
 StakingEscrow ............ {staking_agent.contract_address}
 PolicyManager ............ {policy_agent.contract_address}
 Adjudicator .............. {adjudicator_agent.contract_address} 
     """
 
-    network_payload = f"""
-| Staking |
+    blockchain = f"""    
+| '{blockchain.client.chain_name}' Blockchain Network |
+Gas Price ................ {Web3.fromWei(blockchain.client.gas_price, 'gwei')} Gwei
+Provider URI ............. {blockchain.provider_uri}
+Registry  ................ {registry.filepath}
+    """
+
+    confirmed, pending, inactive = staking_agent.partition_stakers_by_activity()
 
+    staking = f"""
+| Staking |
 Current Period ........... {staking_agent.get_current_period()}
 Actively Staked Tokens ... {NU.from_nunits(staking_agent.get_global_locked_tokens())}
-Published Stakes ......... {staking_agent.get_staker_population()}
-Gas Price ................ {Web3.fromWei(blockchain.client.gas_price, 'gwei')} Gwei
+Stakers population ....... {staking_agent.get_staker_population()}
+   Confirmed ............. {len(confirmed)}
+   Pending confirmation .. {len(pending)}
+   Inactive .............. {len(inactive)}
+
     """
+
+    sep = '-' * 45
+    emitter.echo(sep)
+    emitter.echo(contracts)
+    emitter.echo(sep)
+    emitter.echo(blockchain)
+    emitter.echo(sep)
+    emitter.echo(staking)
+    emitter.echo(sep)
+
+
+def paint_deployer_contract_status(emitter, administrator) -> None:
+
+    blockchain = BlockchainInterfaceFactory.get_interface()
+    token_agent = ContractAgency.get_agent(NucypherTokenAgent, registry=administrator.registry)
+
+    sep = '-' * 45
+    emitter.echo(sep)
+
+    contract_payload = f"""
+
+* Web3 Provider
+====================================================================
+
+Provider URI ............. {blockchain.provider_uri}
+Registry  ................ {administrator.registry.filepath}
+
+* Standard Deployments
+=====================================================================
+
+NucypherToken ........... {token_agent.contract_address}
+    ~ Ethers ............ {blockchain.client.get_balance(token_agent.contract_address)} wei
+    ~ Tokens ............ {token_agent.get_balance(token_agent.contract_address)} wei"""
```

Comment:
Is this supposed to be denominated in `nunits` instead of `wie`?

Reply:
That's from the base PR (#1157). I already left this morning the same comment there.

Reply:
👍